dimod/binary/binary_quadratic_model.py

```diff
@@ -2454,13 +2462,18 @@ def BinaryArray(labels: Union[int, Iterable[Variable]],
     single binary variable.
 
     Args:
-        labels: Either an iterable of variable labels or a number. If a number
-            labels are generated using :class:`uuid.UUID`.
+        labels: Either an iterable of variable labels or the number of required
+            models. If a number, labels are generated using :class:`uuid.UUID`.
         dtype: Data type for the returned binary quadratic models.
 
     Returns:
-        Array of binary quadratic models, each with a single binary variable.
+        Array of :class:`.BinaryQuadraticModel`\ s, each representing a single
+        binary variable.
 
+    Examples:
+        >>> x = dimod.BinaryArray(["x0", "x1"])
+        >>> x[1]
```

Comment:
Again, I think this should just point to symbolic, but if we must put an example here why not just display `x`? Or better yet
```python
>>> x = dimod.BinaryArray(["x0", "x1"])
>>> Q = [[1, 2], [0, 3]]
>>> x.dot(Q).dot(x)
BinaryQuadraticModel({'x0': 1.0, 'x1': 3.0}, {('x1', 'x0'): 2.0}, 0.0, 'BINARY')
```
I don't think this example conveys what this function is useful for very well.

Reply:
I love this idea. 
I wanted to keep the example code as short as possible but do think that because we know this concept is unintuitive it's worth keeping what it does in front of users. If you're okay with the extra bit of space, something like this is much better.

Reply:
I am ok with the extra space, though still mildly prefer pointing everyone to one larger example in a centralized space.